patroni/validator.py

```diff
@@ -397,6 +424,38 @@ def __init__(self, name: str, default: OptionalType[Any] = None) -> None:
         self.default = default
 
 
+class OneOf(object):
+    """Mark that at most one option from a `CaseOptions` can be suplied
+
+    Represents a list of possible configuration options in a given scope, where at most one can actually
+    be provided.
+
+    .. note::
+
+        It should be used together with a :class:`CaseOptions` object, where every field in the :class:`CaseOptions`
+        is something that is :class:`Optional`
+    """
+
+    def __init__(self, *args: Any) -> None:
```

Comment:
Unlike in `Or` here we don't expect anything but strings.
```suggestion
    def __init__(self, *args: str) -> None:
```